components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/core/deployment/CarbonDeploymentSchedulerTask.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.core.deployment;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.deployment.RepositoryListener;
+import org.apache.axis2.deployment.scheduler.SchedulerTask;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * This task takes care of deployment in WSO2 Carbon servers.
+ * <p/>
+ * It will do a deployment synchronization, followed by hot deployment.
+ */
+public class CarbonDeploymentSchedulerTask extends SchedulerTask {
+
+    /**
+     * Indicates whether a Deployment repo update has to be performed.
+     */
+    private static final String REPO_UPDATE_REQUIRED = "repo.update.required";
+
+    private static final Log log = LogFactory.getLog(CarbonDeploymentSchedulerTask.class);
+
+    public CarbonDeploymentSchedulerTask(RepositoryListener listener, AxisConfiguration axisConfig) {
+        super(listener, axisConfig);
+
+        try {
+            axisConfig.addParameter(REPO_UPDATE_REQUIRED, new AtomicBoolean(false));
```

Comment:
When does this become true?

Reply:
In EI this parameter is used to synchronize repositories among tenants. Removing this parameter since we do not have this use case in MI.